public/scripts/instruct-mode.js

```diff
@@ -311,9 +329,9 @@ export const force_output_sequence = {
  * @returns {string} Formatted instruct mode chat message.
  */
 export function formatInstructModeChat(name, mes, isUser, isNarrator, forceAvatar, name1, name2, forceOutputSequence) {
-    let includeNames = isNarrator ? false : power_user.instruct.names;
+    let includeNames = isNarrator ? false : power_user.instruct.names_behavior === names_behavior_types.ALWAYS;
 
-    if (!isNarrator && power_user.instruct.names_force_groups && (selected_group || forceAvatar)) {
+    if (!isNarrator && power_user.instruct.names_behavior === names_behavior_types.FORCE && ((selected_group && name !== name1) || (forceAvatar && name !== name1))) {
```

Comment:
Nice that you added this check for `name !== name1` on the `forceAvatar` **after** I fixed `/sendas` to not include forced avatars if the char is the same as the current chat :D

Though it makes sense to have both, especially for historic purposes. There is no need to enforce name based on forced avatar alone, if the name is the same as the char.